api/STYLE.md

```diff
@@ -25,11 +25,10 @@ In addition, the following conventions should be followed:
 
 * Use [wrapped scalar
   types](https://github.com/google/protobuf/blob/master/src/google/protobuf/wrappers.proto)
-  where there is a real need for the field to have a default value that does not
-  match the proto3 defaults (0/false/""). This should not be done for fields
-  where the proto3 defaults make sense. All things being equal, pick appropriate
-  logic, e.g. enable vs. disable for a `bool` field, such that the proto3
-  defaults work, but only where this doesn't result in API gymnastics.
+  if there is any potential need for a field to have a default value that does not
```

Comment:
AFIK people (self-included =P) stink at predicting this need. 
I'm fine with this as-is, and API folks can review for potential need, but alternately what do you think of just saying "use wrapped scalars" and adding presubmit checks and removing ambiguity?  

cc @markdroth @lizan for thoughts

Reply:
It does seem a bit safer to just always use wrappers. I believe we've stuck to scalars vs wrappers as it neatly implies a default value for scalars, but maybe it would better overall to enforce wrappers everywhere and rely on some annotation (either proto option or comment) that specifies the default?

Doing a backwards compatible change here is obviously not doable without breaking changes, but I'm sure we could enforce it going forward if we so chose?

Reply:
Yeah changing existing fields to wrappers won't be backward compatible. For new fields, if we mandate it for all fields, it might affect performance in term of memory, size on wire etc, but I don't have much insight on it.

Reply:
The huge irony is we had this WKT everywhere originally in v2 xDS and moved away from the policy. The discussion was lost in the https://github.com/envoyproxy/data-plane-api catacombs. I think the biggest objection at the time to using wrappers everywhere is that they can be awkward from a language proto binding perspective, but maybe there were others? I don't recall efficiency coming up, but as we work on xDS scalability, this might be a constant factor that might be big enough to be relevant (e.g. if it's a 10x constant factor).